fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimer.java

```diff
@@ -482,11 +482,11 @@ public enum Counts implements Count {
         /** The number of times the store state cache was unable to return a cached result. */
         STORE_STATE_CACHE_MISS("store info cache miss", false),
         /** The number of record key-value pairs saved. */
-        SAVE_RECORD_KEY("number of record keys saved", false),
+        SAVE_RECORD_KEY("number of record keys saved", false, null, true),
```

Comment:
I think it's true that there are some places where one might want to know some of these sizes on unsuccessful commits (e.g., knowing the byte sizes of indexes vs records when a transaction fails with "transaction too large"), but I think the benefit of only recording bytes written when the write actually goes through to storage is still pretty valuable. In particular, it can help avoid red herrings when investigating the source of writes during, for example, a write hot shard. And if the transaction isn't committed, these bytes here never get persisted to storage (neither the transaction log or the storage server), though it can impact things like the network bandwidth between the FDB client and the commit proxy, or the commit proxies and the resolvers.

Still, I think it made sense to only include `Count`s here, rather than events with times, specifically because we'd want to know the timing for things like `SAVE_RECORD` if we're trying to debug things like transaction timeouts.

There's a part of me that thinks that an enhancement we'd want to handle the case of transactions failing to commit is something like a `FAILED_BYTES_WRITTEN` (and other `FAILED_` metrics). I'm also not quite sure what you'd do with `commit_unknown_result` to indicate maybe-committed bytes. But I think that's an enhancement we'd want to make separately from this PR.

Reply:
I think that makes sense. I wasn't thinking about some of those use cases.
`commit_unknown_result` is probably problematic no matter what, but hopefully those are super rare.
I suspect what we should actually do for failed commits is be more focused, and have those as log info on the exception thrown, and they should have completely different names.